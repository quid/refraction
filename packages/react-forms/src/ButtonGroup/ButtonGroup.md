The `ButtonGroup` component is designed to group two or more buttons into
a visually cohesive widget.

Simple `ButtonGroup` example, without selectable (or activable) buttons:

```js
import { Button } from '@quid/react-forms';

<ButtonGroup>
  <Button>Option 1</Button>
  <Button>Option 2</Button>
  <Button>Option 3</Button>
</ButtonGroup>;
```

The following example, instead, tracks the clicked button, this can be used
as a "radio selector" like component.

To mark a button as selected, simply assign it, as `className`,
the `activeClass` string provided by the `children` render-prop:

```js
import { Button } from '@quid/react-forms';

const buttons = [
  { id: '1', label: 'Option 1' },
  { id: '2', label: 'Option 2' },
  { id: '3', label: 'Option 3' },
];

initialState = { activeId: buttons[0].id };

<ButtonGroup>
  {({ selectedButtonClass }) =>
    buttons.map(({ id, label }) => (
      <Button
        className={state.activeId === id && selectedButtonClass}
        onClick={() => setState({ activeId: id })}
        key={id}
      >
        {label}
      </Button>
    ))
  }
</ButtonGroup>;
```

Here we represent the example above, but this time we'll provide
to `ButtonGroup` the property `disabled`, this will disable all
the buttons inside the component:

```js
import { Button } from '@quid/react-forms';

const buttons = [
  { id: '1', label: 'Option 1' },
  { id: '2', label: 'Option 2' },
  { id: '3', label: 'Option 3' },
];

initialState = { activeId: buttons[0].id };

<ButtonGroup disabled>
  {({ selectedButtonClass }) =>
    buttons.map(({ id, label }) => (
      <Button
        className={state.activeId === id && selectedButtonClass}
        onClick={() => setState({ activeId: id })}
        key={id}
      >
        {label}
      </Button>
    ))
  }
</ButtonGroup>;
```

Example of `ButtonGroup` working with a custom button component:

```js
import { Button } from '@quid/react-forms';

const MyButton = props => <button style={{ color: 'red' }} {...props} />;

<ButtonGroup>
  {({ buttonClass }) => (
    <>
      <MyButton className={buttonClass}>Option 1</MyButton>
      <MyButton className={buttonClass}>Option 2</MyButton>
      <MyButton className={buttonClass}>Option 3</MyButton>
    </>
  )}
</ButtonGroup>;
```
