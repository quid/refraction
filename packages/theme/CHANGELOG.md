# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.25.1](https://github.com/quid/refraction/compare/v1.25.0...v1.25.1) (2019-01-25)

**Note:** Version bump only for package @quid/theme





# [1.24.0](https://github.com/quid/refraction/compare/v1.23.0...v1.24.0) (2019-01-23)


### Bug Fixes

* several build problems that prevented proper usage ([d24bdd1](https://github.com/quid/refraction/commit/d24bdd1))





# [1.23.0](https://github.com/quid/refraction/compare/v1.22.0...v1.23.0) (2019-01-21)


### Features

* add caption text style ([71f4d3f](https://github.com/quid/refraction/commit/71f4d3f))





## [1.21.1](https://github.com/quid/refraction/compare/v1.21.0...v1.21.1) (2019-01-16)


### Bug Fixes

* add pointer cursor to links ([3a4834d](https://github.com/quid/refraction/commit/3a4834d))





# [1.14.0](https://github.com/quid/refraction/compare/v1.13.0...v1.14.0) (2019-01-03)


### Bug Fixes

* theme colors ([7e24c81](https://github.com/quid/refraction/commit/7e24c81))





# [1.11.0](https://github.com/quid/refraction/compare/v1.10.0...v1.11.0) (2018-12-13)


### Bug Fixes

* loosen return type ([0586418](https://github.com/quid/refraction/commit/0586418))





# [1.10.0](https://github.com/quid/refraction/compare/v1.9.3...v1.10.0) (2018-12-13)


### Bug Fixes

* don't apply "normal" text style by default ([775c447](https://github.com/quid/refraction/commit/775c447))


### Features

* export a fonts index.css to include all at once ([668fefb](https://github.com/quid/refraction/commit/668fefb))





## [1.9.1](https://github.com/quid/refraction/compare/v1.9.0...v1.9.1) (2018-12-11)


### Bug Fixes

* properly set JSX pragma at build time ([817eab8](https://github.com/quid/refraction/commit/817eab8))





# [1.9.0](https://github.com/quid/refraction/compare/v1.8.0...v1.9.0) (2018-12-11)


### Bug Fixes

* use Emotion babel macro ([b4e2746](https://github.com/quid/refraction/commit/b4e2746))





# [1.8.0](https://github.com/quid/refraction/compare/v1.7.1...v1.8.0) (2018-12-10)


### Bug Fixes

* get rid of build warnings ([3d5076e](https://github.com/quid/refraction/commit/3d5076e))
* move iconic font to [@quid](https://github.com/quid)/theme ([b2a0a97](https://github.com/quid/refraction/commit/b2a0a97))





## [1.7.1](https://github.com/quid/refraction/compare/v1.7.0...v1.7.1) (2018-12-07)


### Bug Fixes

* textStyles now works reliably ([ece6ff6](https://github.com/quid/refraction/commit/ece6ff6))
* use semibold instead of bold ([5014425](https://github.com/quid/refraction/commit/5014425))





# [1.7.0](https://github.com/quid/refraction/compare/v1.6.0...v1.7.0) (2018-12-07)


### Features

* switch to IBM Plex font family ([7745489](https://github.com/quid/refraction/commit/7745489))





# [1.6.0](https://github.com/quid/refraction/compare/v1.5.3...v1.6.0) (2018-12-07)


### Features

* add current property to identify active theme ([e7ee0e6](https://github.com/quid/refraction/commit/e7ee0e6))





## [1.5.3](https://github.com/quid/refraction/compare/v1.5.1...v1.5.3) (2018-12-06)


### Bug Fixes

* emotion warning ([5ac8686](https://github.com/quid/refraction/commit/5ac8686))





## [1.5.2](https://github.com/quid/refraction/compare/v1.5.1...v1.5.2) (2018-12-06)


### Bug Fixes

* emotion warning ([5ac8686](https://github.com/quid/refraction/commit/5ac8686))





## [1.5.1](https://github.com/quid/refraction/compare/v1.5.0...v1.5.1) (2018-12-06)


### Bug Fixes

* use resolutions to point to microbundle fork ([16819ab](https://github.com/quid/refraction/commit/16819ab))





# [1.5.0](https://github.com/quid/refraction/compare/v1.4.0...v1.5.0) (2018-12-06)


### Bug Fixes

* use testPathPattern rather than runTestsByPath ([2350a80](https://github.com/quid/refraction/commit/2350a80))





# [1.4.0](https://github.com/quid/refraction/compare/v1.3.1...v1.4.0) (2018-12-05)

**Note:** Version bump only for package @quid/theme





# [1.3.0](https://github.com/quid/refraction/compare/v1.2.1...v1.3.0) (2018-12-05)


### Bug Fixes

* properly render CSS ([05f2920](https://github.com/quid/refraction/commit/05f2920))
* textStyles is not part of the theme ([919b026](https://github.com/quid/refraction/commit/919b026))


### Features

* new theme and react-forms packages ([61a44a0](https://github.com/quid/refraction/commit/61a44a0))
