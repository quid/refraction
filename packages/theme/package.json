{
  "name": "@quid/theme",
  "version": "4.11.4",
  "description": "Colors, fonts and other styling related resources, consumable by Emotion",
  "main": "dist/index.js",
  "main:umd": "dist/index.umd.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "start": "microbundle watch",
    "prepare": "microbundle build --jsx React.createElement && flow-copy-source --ignore '{__mocks__/*,*.test}.js' src dist",
    "test": "cd ../.. && yarn test --testPathPattern packages/theme"
  },
  "dependencies": {
    "@emotion/core": "^10.0.10",
    "@emotion/css": "^10.0.4",
    "@emotion/utils": "^0.11.1",
    "color": "^3.1.0",
    "emotion-theming": "^10.0.2"
  },
  "devDependencies": {
    "@emotion/styled": "^10.0.4",
    "flow-copy-source": "^2.0.2",
    "microbundle": "0.12.0-next.6",
    "react": "^16.3.0"
  },
  "peerDependencies": {
    "react": "15||16"
  }
}
