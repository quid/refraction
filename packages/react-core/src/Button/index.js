/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import css from '@emotion/css/macro';
import styled from '@emotion/styled/macro';
import { Link } from 'react-router-dom';
import Color from 'color';
import { withFallback as wf, textStyles } from '@quid/theme';
import Icon from '../Icon';

const OKAY = '#039849';
const WARNING = '#FFCE03';
const HAZARD = '#E61E27';

const HEIGHT = {
  small: 24,
  regular: 32,
};

export type Importance =
  | 'primary'
  | 'secondary'
  | 'okay'
  | 'warning'
  | 'hazard';

export type Size = 'regular' | 'small';

type Props = {
  /** The "color" of the button */
  importance?: Importance,
  /** Make the button background transparent */
  transparent?: boolean,
  /** Button height */
  size?: Size,
  /** What to render inside the button */
  children?: React.Node,
  /** Normal `href` URL, if present, the button will render as anchor (`<a />`) */
  href?: string,
  /** react-router URL, if present, the button will render as `Link` */
  to?: string | Object,
  /** when `true`, the button can't be clicked */
  disabled?: boolean,
  /** type to apply to the button HTMLelement when rendering a "real" button */
  type?: 'button' | 'submit' | 'reset',
  /** onClick callback, called when the element is clicked */
  onClick?: (SyntheticEvent<HTMLElement>) => void,
};

const hoverMod = color =>
  Color(color)
    .darken(0.1)
    .string();
const activeMod = color =>
  Color(color)
    .darken(0.2)
    .string();
const disabledMod = color =>
  Color(color)
    .alpha(0.4)
    .string();

const reset = css`
  border: 0;
  padding: 0;
  background: none;
  cursor: pointer;
  font-size: inherit;
  font-family: inherit;
  text-decoration: none;
  box-sizing: border-box;
  color: inherit;
`;

const Button: React.StatelessFunctionalComponent<Props> & {
  Importance: Importance,
  Size: Size,
} = styled(
  React.forwardRef(
    (
      {
        to,
        href,
        transparent,
        size,
        importance,
        disabled,
        type,
        children,
        onClick,
        ...props
      }: Props,
      ref: React.ElementRef<any>
    ) => {
      let Tag, specificProps;
      if (to && !disabled) {
        Tag = Link;
        specificProps = { to };
      } else if ((href || to) && !disabled) {
        Tag = 'a';
        specificProps = { href };
      } else {
        Tag = 'button';
        specificProps = { disabled, type };
      }

      // FIXME: React doesn't detect when buttons are disabled because
      // of a disabled parent fieldset. With this function we can check it
      // by ourselves and avoid to call `onClick` when it shouldn't.
      // https://github.com/facebook/react/issues/7711
      const isButtonDisabled = button =>
        Tag === 'button' && button.matches(':disabled');

      return (
        <Tag
          {...specificProps}
          {...props}
          onClick={evt =>
            onClick && !isButtonDisabled(evt.currentTarget)
              ? onClick(evt)
              : undefined
          }
          ref={ref}
        >
          {React.Children.map(children, node =>
            ['string', 'number'].includes(typeof node) ? (
              <span>{node}</span>
            ) : (
              node
            )
          )}
        </Tag>
      );
    }
  )
)`
  ${reset};

  ${textStyles('normal', 'bold')};
  display: inline-flex;
  vertical-align: top;
  align-items: center;
  border-radius: 2px;
  height: ${props => HEIGHT[props.size]}px;
  padding-left: 0.769em;
  padding-right: 0.769em;
  transition: padding 0.2s ease-in-out, background 0.2s ease-in-out;
  color: ${wf(props => props.theme.colors.black)};
  border: 1px solid transparent;

  ${wf(props => {
    const variations = {
      primary: props.theme.colors.aqua,
      secondary: props.theme.colors.gray2,
      okay: OKAY,
      warning: WARNING,
      hazard: HAZARD,
    };

    const color = variations[props.importance];

    return css`
      background-color: ${color};
      &:hover,
      &[data-state*='hover'] {
        background-color: ${hoverMod(color)};
      }
      &:active,
      &[data-state*='active'] {
        background-color: ${activeMod(color)};
      }
      ${props.disabled ? '&' : '&:disabled'} {
        &,
        &:hover,
        &:focus,
        &:active {
          background-color: ${disabledMod(color)};
        }
      }
      &:focus:not(:focus-visible) {
        outline: 0;
      }
      &:focus-visible,
      &[data-state*='focus'] {
        outline: 0;
        box-shadow: 0 0 0 0.5px ${props.theme.background}, 0 0 2px 2px ${color};
      }
    `;
  })}

  ${props =>
    props.transparent &&
    css`
      background-color: transparent;
      color: ${props.theme.primary};
    `}


  ${props => (props.disabled ? '&' : '&:disabled')} {
    cursor: default;
    &,
    &:hover,
    &:focus,
    &:active {
      color: ${props => props.theme.disabled};
    }
  }

  ${Icon} {
    display: block;
    line-height: 1;
    font-size: ${props => (props.size === 'small' ? 1 : 1.42)}em;
    margin: 5px -6px;
  }

  ${props =>
    React.Children.count(props.children) > 1 &&
    css`
      ${Icon} {
        display: inline-block;
        position: relative;
        font-size: 1em;
        margin-left: 0;
        bottom: -1px;
        margin-left: 0.35em;
        margin-right: 0;
        &:first-child {
          margin-right: 0.35em;
          margin-left: 0;
        }
        &:last-child {
          margin-left: 0.35em;
          margin-right: 0;
        }
      }
    `}
`;

// $FlowFixMe: https://github.com/facebook/flow/issues/5692
Button.defaultProps = {
  importance: 'secondary',
  size: 'regular',
  type: 'button',
};

// @component
export default Button;
