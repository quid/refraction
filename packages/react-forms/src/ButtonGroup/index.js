/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import styled from '@emotion/styled/macro';
import { ClassNames } from '@emotion/core';
import { withFallback as wf } from '@quid/theme';
import nanoid from 'nanoid';
import Button from '../Button';

// prettier-ignore
const COLOR = wf(({ theme, disabled }) =>
  theme.current === 'light'
    ? (disabled
      ? theme.colors.gray3
      : theme.colors.gray5
    )
    : (disabled
      ? theme.colors.gray3
      : theme.colors.gray2
    )
);
// prettier-ignore
const ACTIVE = wf(({ theme, disabled }) =>
  theme.current === 'light'
    ? (disabled
      ? theme.colors.gray4
      : theme.colors.gray7
    )
    : (disabled
      ? theme.colors.gray2
      : theme.colors.white
    )
);
// prettier-ignore
const BG_1 = wf(({ theme, disabled }) =>
  theme.current === 'light'
    ? (disabled
      ? theme.colors.gray0
      : theme.colors.gray1
    )
    : (disabled
      ? theme.colors.gray6
      : theme.colors.gray5
    )
);
// prettier-ignore
const BORDER = wf(({ theme, disabled }) =>
  theme.current === 'light'
    ? (disabled
      ? theme.colors.gray2
      : theme.colors.gray5
    )
    : (disabled
      ? theme.colors.gray4
      : theme.colors.gray2
    )
);
const BG_2 = wf(props =>
  props.theme.current === 'light'
    ? props.theme.colors.gray2
    : props.theme.colors.gray4
);

const Fieldset = styled.fieldset`
  all: unset;
`;

const Container = styled.div`
  display: inline-flex;
  position: relative;
  align-items: stretch;

  &::before {
    content: '';
    position: absolute;
    isolation: isolate;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    border: 1px solid ${BORDER};
    border-radius: 2px;
    pointer-events: none;
  }

  ${Button}, .${props => props.buttonClass} {
    border-radius: 0;
    background-color: transparent;
    color: ${COLOR};
    box-shadow: inset 0 0 0 0.5px ${BORDER};
    border: 0;

    &:hover {
      background-color: ${BG_1};
    }
    &:active {
      background-color: ${BG_2};
    }
    &:disabled {
      &,
      &:hover,
      &:active {
        background-color: transparent;
      }
    }

    &.${props => props.selectedButtonClass} {
      &,
      &:disabled {
        background-color: ${BG_1};
        color: ${ACTIVE};
      }
    }

    &:focus-visible {
      z-index: 1;
      background-color: ${BG_1};
      box-shadow: inset 0 0 0 1px ${BORDER},
        0 0 1px 0.5px ${wf(props => props.theme.background)},
        0 0 2px 2px ${wf(props => props.theme.colors.gray7)};
    }

    &:first-of-type {
      border-top-left-radius: 2px;
      border-bottom-left-radius: 2px;
    }
    &:last-of-type {
      border-top-right-radius: 2px;
      border-bottom-right-radius: 2px;
    }
  }
`;

type Props = {
  /**
   * Either a list of buttons, or a function which takes as argument an object
   * with two class names (buttonClass,  and selectedButtonClass), and returns a list of buttons
   */
  children:
    | React.Node
    | (({ buttonClass: string, selectedButtonClass: string }) => React.Node),
  /** Class used to target custom button components */
  buttonClass?: string,
  /** The class name used to target the selected/activated buttons, by default is a random string */
  selectedButtonClass?: string,
  /** Set this to `true` to disable all the buttons inside the component */
  disabled?: boolean,
};

const ButtonGroup = ({
  children,
  buttonClass = nanoid(),
  selectedButtonClass = nanoid(),
  disabled,
  ...props
}: Props) => (
  <Fieldset disabled={disabled}>
    <Container
      buttonClass={buttonClass}
      selectedButtonClass={selectedButtonClass}
      disabled={disabled}
    >
      <ClassNames>
        {({ css }) =>
          typeof children === 'function'
            ? children({ buttonClass, selectedButtonClass })
            : children
        }
      </ClassNames>
    </Container>
  </Fieldset>
);

export default ButtonGroup;
