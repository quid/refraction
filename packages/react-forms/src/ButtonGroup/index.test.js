/**
 * Copyright (c) Quid, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
// @flow
import * as React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from '@quid/theme';
import Button from '../Button';
import ButtonGroup from '.';

jest.mock('../Button', () => 'button');
jest.mock('nanoid', () => () => 'NOT_VERY_RANDOM_STRING');

it('renders a 2 buttons ButtonGroup', () => {
  expect(
    mount(
      <ButtonGroup>
        <Button>a</Button>
        <Button>b</Button>
      </ButtonGroup>
    )
  ).toMatchSnapshot();
});

it('renders the dark theme version', () => {
  expect(
    mount(
      <ThemeProvider theme="dark">
        <ButtonGroup>
          <Button>a</Button>
          <Button>b</Button>
        </ButtonGroup>
      </ThemeProvider>
    )
  ).toMatchSnapshot();
});

it('renders a 3 buttons ButtonGroup', () => {
  expect(
    mount(
      <ButtonGroup>
        <Button>a</Button>
        <Button>b</Button>
        <Button>c</Button>
      </ButtonGroup>
    ).find('button')
  ).toHaveLength(3);
});

it('renders an active button', () => {
  expect(
    mount(
      <ButtonGroup>
        {({ selectedButtonClass }) => (
          <Button className={selectedButtonClass}>a</Button>
        )}
      </ButtonGroup>
    ).find('button')
  ).toMatchInlineSnapshot(`
    <button
      className="NOT_VERY_RANDOM_STRING"
    >
      a
    </button>
  `);
});

it('renders a disabled ButtonGroup', () => {
  expect(
    mount(
      <ButtonGroup disabled>
        <Button disabled>a</Button>
      </ButtonGroup>
    )
      .find('fieldset')
      .prop('disabled')
  ).toBe(true);
});

it('renders a dark disabled ButtonGroup (without throwing errors)', () => {
  expect(
    mount(
      <ThemeProvider theme="dark">
        <ButtonGroup disabled>
          <Button disabled>a</Button>
        </ButtonGroup>
      </ThemeProvider>
    )
      .find('fieldset')
      .prop('disabled')
  ).toBe(true);
});
