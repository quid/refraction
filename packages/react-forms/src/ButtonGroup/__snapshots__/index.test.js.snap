// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a 2 buttons ButtonGroup 1`] = `
.emotion-2 {
  all: unset;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-0::before {
  content: '';
  position: absolute;
  isolation: isolate;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border: 1px solid #49535A;
  border-radius: 2px;
  pointer-events: none;
}

.emotion-0 button,
.emotion-0 .NOT_VERY_RANDOM_STRING {
  border-radius: 0;
  background-color: transparent;
  color: #49535A;
  box-shadow: inset 0 0 0 0.5px #49535A;
  border: 0;
}

.emotion-0 button:hover,
.emotion-0 .NOT_VERY_RANDOM_STRING:hover {
  background-color: #E3E6E8;
}

.emotion-0 button:active,
.emotion-0 .NOT_VERY_RANDOM_STRING:active {
  background-color: #C7CCD1;
}

.emotion-0 button:disabled,
.emotion-0 .NOT_VERY_RANDOM_STRING:disabled,
.emotion-0 button:disabled:hover,
.emotion-0 .NOT_VERY_RANDOM_STRING:disabled:hover,
.emotion-0 button:disabled:active,
.emotion-0 .NOT_VERY_RANDOM_STRING:disabled:active {
  background-color: transparent;
}

.emotion-0 button.NOT_VERY_RANDOM_STRING,
.emotion-0 .NOT_VERY_RANDOM_STRING.NOT_VERY_RANDOM_STRING,
.emotion-0 button.NOT_VERY_RANDOM_STRING:disabled,
.emotion-0 .NOT_VERY_RANDOM_STRING.NOT_VERY_RANDOM_STRING:disabled {
  background-color: #E3E6E8;
  color: #1B1F22;
}

.emotion-0 button:focus-visible,
.emotion-0 .NOT_VERY_RANDOM_STRING:focus-visible {
  z-index: 1;
  background-color: #E3E6E8;
  box-shadow: inset 0 0 0 1px #49535A,0 0 1px 0.5px #FFFFFF,0 0 2px 2px #1B1F22;
}

.emotion-0 button:first-of-type,
.emotion-0 .NOT_VERY_RANDOM_STRING:first-of-type {
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
}

.emotion-0 button:last-of-type,
.emotion-0 .NOT_VERY_RANDOM_STRING:last-of-type {
  border-top-right-radius: 2px;
  border-bottom-right-radius: 2px;
}

<Fieldset>
  <fieldset
    className="emotion-2 emotion-3"
  >
    <Container
      buttonClass="NOT_VERY_RANDOM_STRING"
      selectedButtonClass="NOT_VERY_RANDOM_STRING"
    >
      <div
        className="emotion-0 emotion-1"
      >
        <ForwardRef(render)>
          <button>
            a
          </button>
          <button>
            b
          </button>
        </ForwardRef(render)>
      </div>
    </Container>
  </fieldset>
</Fieldset>
`;

exports[`renders the dark theme version 1`] = `
.emotion-2 {
  all: unset;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-0::before {
  content: '';
  position: absolute;
  isolation: isolate;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border: 1px solid #C7CCD1;
  border-radius: 2px;
  pointer-events: none;
}

.emotion-0 button,
.emotion-0 .NOT_VERY_RANDOM_STRING {
  border-radius: 0;
  background-color: transparent;
  color: #C7CCD1;
  box-shadow: inset 0 0 0 0.5px #C7CCD1;
  border: 0;
}

.emotion-0 button:hover,
.emotion-0 .NOT_VERY_RANDOM_STRING:hover {
  background-color: #49535A;
}

.emotion-0 button:active,
.emotion-0 .NOT_VERY_RANDOM_STRING:active {
  background-color: #6C7983;
}

.emotion-0 button:disabled,
.emotion-0 .NOT_VERY_RANDOM_STRING:disabled,
.emotion-0 button:disabled:hover,
.emotion-0 .NOT_VERY_RANDOM_STRING:disabled:hover,
.emotion-0 button:disabled:active,
.emotion-0 .NOT_VERY_RANDOM_STRING:disabled:active {
  background-color: transparent;
}

.emotion-0 button.NOT_VERY_RANDOM_STRING,
.emotion-0 .NOT_VERY_RANDOM_STRING.NOT_VERY_RANDOM_STRING,
.emotion-0 button.NOT_VERY_RANDOM_STRING:disabled,
.emotion-0 .NOT_VERY_RANDOM_STRING.NOT_VERY_RANDOM_STRING:disabled {
  background-color: #49535A;
  color: #FFFFFF;
}

.emotion-0 button:focus-visible,
.emotion-0 .NOT_VERY_RANDOM_STRING:focus-visible {
  z-index: 1;
  background-color: #49535A;
  box-shadow: inset 0 0 0 1px #C7CCD1,0 0 1px 0.5px #2E3338,0 0 2px 2px #1B1F22;
}

.emotion-0 button:first-of-type,
.emotion-0 .NOT_VERY_RANDOM_STRING:first-of-type {
  border-top-left-radius: 2px;
  border-bottom-left-radius: 2px;
}

.emotion-0 button:last-of-type,
.emotion-0 .NOT_VERY_RANDOM_STRING:last-of-type {
  border-top-right-radius: 2px;
  border-bottom-right-radius: 2px;
}

<Fieldset>
  <fieldset
    className="emotion-2 emotion-3"
  >
    <Container
      buttonClass="NOT_VERY_RANDOM_STRING"
      selectedButtonClass="NOT_VERY_RANDOM_STRING"
    >
      <div
        className="emotion-0 emotion-1"
      >
        <ForwardRef(render)>
          <button>
            a
          </button>
          <button>
            b
          </button>
        </ForwardRef(render)>
      </div>
    </Container>
  </fieldset>
</Fieldset>
`;
